xcodeproj/internal/filelists.bzl

```diff
@@ -0,0 +1,46 @@
+"""Module containing functions dealing with file lists."""
+
+def _write(*, ctx, rule_name, name, files):
+    if files == None:
+        files = depset()
+
+    args = ctx.actions.args()
+    args.use_param_file("%s", use_always = True)
+    args.set_param_file_format("multiline")
+    args.add_all(files, expand_directories = False)
+
+    output = ctx.actions.declare_file("{}-{}.filelist".format(rule_name, name))
+
+    ctx.actions.run_shell(
+        command = """
+if [[ $OSTYPE == darwin* ]]; then
+  cp -c \"$1\" \"$2\"
+else
+  cp \"$1\" \"$2\"
+fi
+""",
+        arguments = [
+            args,
+            output.path,
```

Comment:
Probably overkill but we could also put `output` in another `Args` object.

Reply:
I don't see the benefit. The existing `Args` object is used in order to get the `.params` file from it, and to prevent expanding the `files` `Depset`. `output` on the other-hand doesn't have that issue, and putting it in an `Args` object just to get unwrapped will be unnecessary overhead, right?

Reply:
Ahh, so it reduces memory usage, for when this action isn't run. I didn't think of that. I'll make that change.